File: 10-Java-Basic-Web/Calculator/src/main/java/com/softuni/controller/HomeController.java
package com.softuni.controller;

import com.softuni.bindingModel.CalculatorBindingModel;
import com.softuni.entity.Calculator;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;

@Controller
public class HomeController {
	@GetMapping("/")
	public String index(Model model) {
		model.addAttribute("operator", "+");
		model.addAttribute("view", "views/calculatorForm");
		return "base-layout";
	}

	@PostMapping("/")
    public  String calculate(CalculatorBindingModel validator, Model model) {
        double numberOne;
        double numberTwo;

	    try {
	        numberOne = Double.parseDouble(validator.getLeftOperand());
        } catch (NumberFormatException ex) {
	        numberOne = 0;
        }

        try {
	        numberTwo = Double.parseDouble(validator.getRightOperand());
        } catch (NumberFormatException ex) {
	        numberTwo = 0;
        }

        Calculator calculator = new Calculator(numberOne, numberTwo, validator.getOperator());

        double result = calculator.CalculateResult();

        String resultStr = String.valueOf(result);
        if (validator.getOperator().equals("/") && numberTwo == 0) resultStr = "Error!";
        if (validator.getOperator().equals("mod") && numberTwo == 0) resultStr = "Error!";

        model.addAttribute("leftOperand", calculator.getLeftOperand());
        model.addAttribute("operator", calculator.getOperator());
        model.addAttribute("rightOperand", calculator.getRightOperand());
        model.addAttribute("result", resultStr);
	    model.addAttribute("view", "views/calculatorForm");

	    return "base-layout";
    }
}

File: 12-CSharp-ASP.NET-MVC-Overview-Exercises/01. Calculator/Calculator-CSharp/Controllers/HomeController.cs
using Calculator_CSharp.Models;
using System;
using System.Web.Mvc;

namespace Calculator_CSharp.Controllers
{
    public class HomeController : Controller
    {
        public ActionResult Index(Calculator calulator)
        {
            return View(calulator);
        }

        [HttpPost]
        public ActionResult Calculate(Calculator calculator)
        {
            calculator.Result = CalculateResult(calculator);

            return RedirectToAction("Index", calculator);
        }

        private decimal CalculateResult(Calculator calculator)
        {
            decimal result = 0m;

            switch (calculator.Operator)
            {
                case "+":
                    result = calculator.LeftOperand + calculator.RightOperand;
                    break;

                case "-":
                    result = calculator.LeftOperand - calculator.RightOperand;
                    break;

                case "*":
                    result = calculator.LeftOperand * calculator.RightOperand;
                    break;

                case "/":
                    if (calculator.RightOperand == 0) return result;
                    result = calculator.LeftOperand / calculator.RightOperand;
                    break;

                case "mod":
                    if (calculator.RightOperand == 0) return result;
                    result = calculator.LeftOperand % calculator.RightOperand;
                    break;

                case "pow":
                    result = (decimal)Math.Pow((double)calculator.LeftOperand, (double)calculator.RightOperand);
                    break;

                case "And":
                    result = (int)calculator.LeftOperand & (int)calculator.RightOperand;
                    break;

                case "Or":
                    result = (int)calculator.LeftOperand | (int)calculator.RightOperand;
                    break;

                case "Xor":
                    result = (int)calculator.LeftOperand ^ (int)calculator.RightOperand;
                    break;

                case "Not":
                    result = ~(int)calculator.LeftOperand;
                    break;

                case "ShiftLeft":
                    result = (int)calculator.LeftOperand << (int)calculator.RightOperand;
                    break;

                case "ShiftRight":
                    result = (int)calculator.LeftOperand >> (int)calculator.RightOperand;
                    break;

                case "What is X% of Y":
                    result = calculator.LeftOperand / 100 * calculator.RightOperand;
                    break;

                case "X is what % of Y":
                    if (calculator.RightOperand != 0)
                    {
                        result = calculator.LeftOperand / calculator.RightOperand * 100;
                    }
                    break;

                case "√X":
                    result = (decimal)Math.Sqrt((double)calculator.LeftOperand);
                    break;

                case "X!":
                    result = CalcFactorial((int)calculator.LeftOperand);
                    break;
            }

            return result;
        }

        public static int CalcFactorial(int number)
        {
            int factorial = 1;

            while (number > 1)
            {
                factorial *= number;
                number--;
            }

            return factorial;
        }
    }
}